src/lightning/pytorch/strategies/ddp.py

```diff
@@ -273,7 +273,7 @@ def optimizer_step(
     def configure_ddp(self) -> None:
         log.debug(f"{self.__class__.__name__}: configuring DistributedDataParallel")
         assert isinstance(self.model, (pl.LightningModule, _LightningPrecisionModuleWrapperBase))
-        self.model = self._setup_model(_LightningModuleWrapperBase(self.model))
```

Comment:
This is the key change and motivation in this PR. By removing the wrapper, an inspection of the module becomes easier and logic is less error-prone.